leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/RegisterResource.java

```diff
@@ -254,9 +254,9 @@ private Identity extractIdentity(CoapExchange exchange) {
                 return Identity.rpk(peerAddress, publicKey);
             } else if (senderIdentity instanceof X500Principal) {
                 // Extract common name
-                Matcher endpointMatcher = Pattern.compile("CN=.*?,").matcher(senderIdentity.getName());
+                Matcher endpointMatcher = Pattern.compile("CN=(.*?)(,|$)").matcher(senderIdentity.getName());
                 if (endpointMatcher.find()) {
-                    String x509CommonName = endpointMatcher.group().substring(3, endpointMatcher.group().length() - 1);
+                    String x509CommonName = endpointMatcher.group(1);
```

Comment:
There is the `leshan-core` module which both *client* and *server* depend on and which contains a `util` package.
I can image putting this functionality there instead ...